mixnode/src/commands/upgrade.rs

```diff
@@ -244,15 +245,87 @@ fn patch_010_upgrade(
 
     let upgraded_config = config.with_custom_version(to_version.to_string().as_ref());
 
-    upgraded_config.save_to_file(None).unwrap_or_else(|err| {
-        eprintln!("failed to overwrite config file! - {:?}", err);
-        print_failed_upgrade(&config_version, &to_version);
-        process::exit(1);
-    });
+    upgraded_config.save_to_file(None).map_err(|err| {
+        (
+            to_version.clone(),
+            format!("failed to overwrite config file! - {:?}", err),
+        )
+    })?;
 
     print_successful_upgrade(config_version, to_version);
 
-    upgraded_config
+    Ok(upgraded_config)
+}
+
+fn patch_0_10_2_upgrade(
+    config: Config,
+    _matches: &ArgMatches,
+    config_version: &Version,
+    package_version: &Version,
+) -> Result<Config, (Version, String)> {
+    #[derive(Deserialize)]
+    struct OldNodeDescription {
+        name: String,
+        description: String,
+        link: String,
+    }
+    let to_version = package_version;
+    let id = config.get_id();
+    let config_path = Config::default_config_directory(&id);
+
+    print_start_upgrade(&config_version, &to_version);
+
+    let upgraded_config = config.with_custom_version(to_version.to_string().as_ref());
+
+    upgraded_config.save_to_file(None).map_err(|err| {
+        (
+            to_version.clone(),
+            format!("failed to overwrite config file! - {:?}", err),
+        )
+    })?;
+
+    let description_file_path: PathBuf = [config_path.to_str().unwrap(), DESCRIPTION_FILE]
+        .iter()
+        .collect();
+    // If the description file already exists, upgrade it
+    if description_file_path.is_file() {
+        let description_content =
+            fs::read_to_string(description_file_path.clone()).map_err(|err| {
+                (
+                    to_version.clone(),
+                    format!("failed to read description file! - {:?}", err),
+                )
+            })?;
+        let old_description: OldNodeDescription =
+            toml::from_str(&description_content).map_err(|err| {
+                (
+                    to_version.clone(),
+                    format!("failed to deserialize description content! - {:?}", err),
+                )
+            })?;
+        let new_description = NodeDescription {
+            name: old_description.name,
+            description: old_description.description,
+            link: old_description.link,
+            ..Default::default()
+        };
+        let description_toml = toml::to_string(&new_description).map_err(|err| {
```

Comment:
Here you can just use `NodeDescription::save_to_file` which will do toml serialization for you

Reply:
Will do.